### PR TITLE
feat: Add requestPage (component, container)

### LINKED ISSUE #67: [frontend] Request page component

### PR DESCRIPTION
**Pull request by mss8087**
### What is the issue this pull request related with?

resolved #67, #68 

### Your checklist for this pull request
🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [X] Code builds correctly
- [X] Created tests which fail without the change (if possible)
- [X] All tests passing
- [X] Extended the README / documentation, if necessary

### Description
Please describe your pull request.

💔Thank you!

### CODE REVIEW

File: frontend/src/components/request/request.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import styled from 'styled-components';
+import Button from '../../components/common/Button/Button';
+const RequestBlock = styled.div``;
+
+Request.propTypes = {
+  //TODO
+};
+
+function onCheckboxClickHandler(name, location) {
```
Comment (zeroFruit) [inline frontend/src/components/request/request.js]: Please move this function inside `Request` component.

```javascript
function Request() {
  const onCheckboxClick = (name, location ) => { ... }
  ...
  return <></>
}
```

File: frontend/src/components/request/request.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+      </div>
+
+      <div data-testid="to">
+        <h2>To</h2>
+        {toData}
+      </div>
+
+      <div data-testid="minimumPassenger">
+        <h2>Minimum Passenger</h2>
+        {minPassenger}
```
Comment (zeroFruit) [inline frontend/src/components/request/request.js]: please change this code

```javascript
{fromData}
```

into

```javascript
{
  fromList && fromList.map(location => 
    Checkbox('From', location);
  );
}
```

inline code + change to an arrow function

File: frontend/src/components/request/request.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import styled from 'styled-components';
+import Button from '../../components/common/Button/Button';
+const RequestBlock = styled.div``;
+
+Request.propTypes = {
+  //TODO
```
Comment (zeroFruit) [inline frontend/src/components/request/request.js]: propTypes are important so please specify the type of props.

File: frontend/src/components/request/Checkbox/Checkbox.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Checkbox from '@material-ui/core/Checkbox';
+import PropTypes from 'prop-types';
+
+CheckboxLabels.propTypes = {
+  name: PropTypes.string,
+  value: PropTypes.string,
+  onClick: PropTypes.func,
+  checked: PropTypes.bool,
+};
```
Comment (zeroFruit) [inline frontend/src/components/request/Checkbox/Checkbox.js]: I think all fields are `required` so changed it to `~.isRequired`

File: frontend/src/components/request/request.js [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    ) {
+      onClickRequest({
+        userId: user.id,
+        from: from,
+        to: to,
+        minimumPassenger: minPassenger,
+      });
+    } else {
+      window.alert('Please Check!');
+    }
```
Comment (zeroFruit) [inline frontend/src/components/request/request.js]: if you are checking `user === null || from === null || to === null` case first we can remove `else` statement.

So fixing like as following:

```
if (user === null || ... ) {
  window.alert(...)
  return
}
<do something...>
```

would be more clean!

File: frontend/src/components/request/request.js [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+            (name === 'to' && to === value) ||
+            (name === 'minimumPassenger' && minPassenger === value)
+          }
+        />
+      );
+    };
+
+  if (!user) {
+    return null;
+  }
```
Comment (zeroFruit) [inline frontend/src/components/request/request.js]: please move this logic into container!

File: frontend/src/components/request/request.js [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+          value={value}
+          onClick={onCheckboxClickHandler(name, value)}
+          checked={
+            (name === 'from' && from === value) ||
+            (name === 'to' && to === value) ||
+            (name === 'minimumPassenger' && minPassenger === value)
+          }
+        />
+      );
+    };
```
Comment (zeroFruit) [inline frontend/src/components/request/request.js]: I think `MakeCheckBox` function has too much responsibility.

I suggest removing this function and separate responsibility:

```javascript
{fromList && fromList.map(MakeCheckbox('from', from, to, minPassenger))}
```

into

```javascript
{fromList && fromList.map(value => (
  <Checkbox
          key={value}
          name={'from'}
          value={value}
          onClick={onCheckboxClickHandler(name, value)}
          checked={value !== from}
        />
))}
```

Comment (zeroFruit) [reply frontend/src/components/request/request.js]: If we change like this I think we don't need `onCheckboxClickHandler` function.

So code would be like this

from 

```javascript
onClick={onCheckboxClickHandler(name, value)}
```

into

```javascript
onClick={() => setFrom(value)}
```

something like this

Comment (mss8087) [reply frontend/src/components/request/request.js]: if I change the onClick function like that, We couldn't cancel the checkbox. Is it okay?

File: frontend/src/components/request/request.js [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+import Checkbox from './Checkbox/Checkbox';
+import PropTypes from 'prop-types';
+const RequestBlock = styled.div``;
+
+Request.propTypes = {
+  user: PropTypes.object,
+  fromList: PropTypes.array,
+  toList: PropTypes.array,
+  minimumPassenger: PropTypes.array,
+  onClickRequest: PropTypes.func,
```
Comment (zeroFruit) [inline frontend/src/components/request/request.js]: please specify `isRequired`

File: frontend/src/contrainers/requestContainer/requestContainer.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+  //TODO: it should be loaded from DB
+  const fromList = ['SNU station', 'Nakseongdae station', 'Nokdu Street'];
+  const toList = ['301 building', 'Student Center', 'Dormitory three-way'];
+  const minimumPassenger = ['2', '3', '4'];
+
+  const onClickRequest = useCallback(
+    payload => {
+      dispatch(requestCarpool(payload));
+      history.push('/waiting');
+    },
```
Comment (zeroFruit) [inline frontend/src/contrainers/requestContainer/requestContainer.js]: for clarity please change `payload` => `({ userId, from, to, minimumPassenger })`

File: frontend/src/pages/RequestPage/RequestPage.js [hunk 10]
```diff
@@ -1,7 +1,12 @@
 import React from 'react';
 import Main from '../_layouts/Main';
+import RequestContainer from 'contrainers/requestContainer/requestContainer';
```
Comment (zeroFruit) [inline frontend/src/pages/RequestPage/RequestPage.js]: please change it
```
import RequestContainer from 'contrainers/RequestContainer'
```

**Pull request merged**